source/standalone/workflows/ray/grok_cluster_with_kubectl.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (c) 2022-2024, The Isaac Lab Project Developers.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import argparse
+import os
+import re
+import subprocess
+import threading
+import time
+from concurrent.futures import ThreadPoolExecutor, as_completed
+
+"""
+This script requires that kubectl is installed and KubeRay was used to create the cluster.
+
+Creates a config file containing ``name: <NAME> address: http://<IP>:<PORT>`` on
+a new line for each cluster.
+
+Usage:
+
+.. code-block:: bash
+
+
+    ./isaaclab.sh -p source/standalone/workflows/ray/grok_cluster_with_kubectl.py
+
+    # For options, supply -h arg
+"""
+
+
+def get_pods(namespace: str = "default") -> list[tuple]:
+    cmd = ["kubectl", "get", "pods", "-n", namespace, "--no-headers"]
+    output = subprocess.check_output(cmd).decode()
+    pods = []
+    for line in output.strip().split("\n"):
+        fields = line.split()
+        pod_name = fields[0]
+        status = fields[2]
+        pods.append((pod_name, status))
+    return pods
+
+
+def get_clusters(pods: list, cluster_name_prefix: str) -> set:
+    clusters = set()
+    # Modify regex pattern to match the entire structure including `-head` or `-worker`
+    for pod_name, _ in pods:
+        match = re.match(r"(" + re.escape(cluster_name_prefix) + r"[-\w]+)", pod_name)
+        if match:
+            clusters.add(match.group(1).split("-head")[0].split("-worker")[0])
+    return sorted(clusters)
+
+
+def check_clusters_running(pods: list, clusters: set) -> bool:
+    clusters_running = True
+    for cluster in clusters:
+        cluster_pods = [p for p in pods if p[0].startswith(cluster)]
+        total_pods = len(cluster_pods)
+        running_pods = len([p for p in cluster_pods if p[1] == "Running"])
+        if running_pods != total_pods:
+            clusters_running = False
+            break
+    return clusters_running
```

Comment:
If clusters is empty, this will return `True`. You might want to assume they aren't running and set to True only if they are?